pkg/lib/cockpit-components-select.jsx

```diff
@@ -20,125 +20,184 @@
                 // if the item didn't change, don't do anything
-                if (elementData === this.state.currentData)
+                if (elementData === this.props.selected) {
                     return;
-                this.setState({ currentData: elementData });
-                if (this.props.onChange)
+                }
+                if (this.props.onChange) {
                     this.props.onChange(elementData);
+                }
             } else {
                 this.setState({ open: !this.state.open });
             }
-        },
-        render: function() {
-            var self = this;
-            var currentValue;
-
-            var listItems = React.Children.map(this.props.children, function(itm) {
-                var data = ('data' in itm.props) ? itm.props.data : undefined;
-                // we need to have some kind of value
-                var value = (itm.props.children !== undefined) ? itm.props.children : textForUndefined;
-                if (data === self.state.currentData)
-                    currentValue = value;
-                // if there's no initial value, use the first one
-                else if (!self.props.initial && currentValue === undefined)
+        }
+
+        render() {
+            const self = this;
+            let currentValue;
+
+            const getItemData = (itm) => (('data' in itm.props) ? itm.props.data : undefined);
+            const getItemValue = (itm) => ((itm.props.children !== undefined) ? itm.props.children : textForUndefined);
+
+            const children = this.props.children;
+            const listItems = React.Children.map(children, function(itm) {
+                const data = getItemData(itm);
+                const value = getItemValue(itm);
+                if (data === self.props.selected) {
                     currentValue = value;
+                }
                 return <li data-value={value} data-data={data}>{itm}</li>;
             });
-            var classes = "btn-group bootstrap-select dropdown";
-            if (this.state.open)
+            // select first item if not found via selected prop
+            if (!currentValue && React.Children.count(children) > 0) {
+                currentValue = getItemValue(React.Children.toArray(children)[0]);
+            }
+
+            let classes = "btn-group bootstrap-select dropdown";
+            if (this.state.open) {
                 classes += " open";
-            var buttonClasses = "btn btn-default dropdown-toggle";
-            if (this.props.enabled === false)
+            }
+            let buttonClasses = "btn btn-default dropdown-toggle";
+            if (this.props.enabled === false) {
                 buttonClasses += " disabled";
+            }
 
             return (
-                <div className={classes} onClick={this.clickHandler} id={this.props.id}>
+                <div className={classes} onClick={this.clickHandler.bind(this)} id={this.props.id}>
```

Comment:
Let's not unnecessarily re-bind `clickHandler()` with every `render()` call. Bind it in the constructor instead.